synapse/datamodel.py

```diff
@@ -1119,6 +1190,26 @@ def addFormProp(self, formname, propname, tdef, info):
 
     def _addFormProp(self, form, name, tdef, info):
 
+        if tdef is None:
+
+            # check if there was an already declared interface def
+            if (prop := form.prop(name)) is not None:
+                tdef = (prop.type.name, prop.type.info)
+                for name, valu in prop.info.items():
+                    info.setdefault(name, valu)
```

Comment:
Currently when adding interfaces we skip calling `_addFormProp` if the form already has a prop with that name, might to move this or adjust some other logic to get this working how we want it

Reply:
moved into a TODO that we can carve out into a new PR :+1: